### DISCUSSION TITLE
The future of Schlange (german python):  Translations for match/case and name convention

### DISCUSSION THREAD

**Discussion by actopozipc** [actopozipc/German-Python-Interpreter#8] [General]
Hey guys :) I got a lot of messages and suggestions for improvement and wanted to ask a few questions here.
1. Since Python 3.10, there is a switch case version for python called match case. Right now, Schlange does not cover this. Any ideas for possible translations? :)
The upper-lower case is not very consistent at the moment, I would start by capitalizing all nouns and all exceptions which will be written in CamelCase if there isn't a full word translation for it. Is that a good thing or are there better ideas?

**Comment by AntonPieper** [comment]
For match case, maybe "abgleichen" and "Fall"? however this would introduce a capital letter where previously was none

**Reply by actopozipc** [reply]
I am not sure anyway for the upper- and lower- cases. Rückkehr for return is for example written uppercase. More pythonian would be to write everything lowercase beside Wahr and Falsch

**Reply by actopozipc** [reply]
Maybe lets call it switch like in every other language and translate it with "wechsel" or "Schalter"?

**Reply by AntonPieper** [reply]
I think "wechsel" implies that the variable changes, which is not the case. "Schalter" sound more like lever to me. So with these keywords the python program back-translates to:
```py
change variable:
    lever "a": pass
```

**Reply by actopozipc** [reply]
I think I have a great idea: "Übereinstimmung" for match and "Fall" for case sounds neat, doesnt it?
